packages/rum-recorder/src/boot/rumRecorderPublicApi.spec.ts

```diff
@@ -28,41 +28,95 @@ describe('makeRumRecorderPublicApi', () => {
-      rumRecorderPublicApi.init(DEFAULT_INIT_CONFIGURATION)
-      expect(startRumSpy).toHaveBeenCalled()
-    })
+  describe('boot', () => {
+    describe('when tracking views automatically', () => {
+      it('starts RUM when init is called', () => {
+        expect(startRumSpy).not.toHaveBeenCalled()
+        rumRecorderPublicApi.init(DEFAULT_INIT_CONFIGURATION)
+        expect(startRumSpy).toHaveBeenCalled()
+      })
 
-    it('starts recording when init() is called', () => {
-      expect(startRecordingSpy).not.toHaveBeenCalled()
-      rumRecorderPublicApi.init(DEFAULT_INIT_CONFIGURATION)
-      expect(startRecordingSpy).toHaveBeenCalled()
-    })
+      it('starts recording when init() is called', () => {
+        expect(startRecordingSpy).not.toHaveBeenCalled()
+        rumRecorderPublicApi.init(DEFAULT_INIT_CONFIGURATION)
+        expect(startRecordingSpy).toHaveBeenCalled()
+      })
 
-    it('does not start recording when calling init() with manualSessionReplayRecordingStart: true', () => {
-      rumRecorderPublicApi.init({ ...DEFAULT_INIT_CONFIGURATION, manualSessionReplayRecordingStart: true })
-      expect(startRecordingSpy).not.toHaveBeenCalled()
+      it('does not start recording when calling init() with manualSessionReplayRecordingStart: true', () => {
+        rumRecorderPublicApi.init({ ...DEFAULT_INIT_CONFIGURATION, manualSessionReplayRecordingStart: true })
+        expect(startRecordingSpy).not.toHaveBeenCalled()
+      })
+
+      it('does not start recording when calling init() with the feature "postpone_start_recording"', () => {
+        rumRecorderPublicApi.init({
+          ...DEFAULT_INIT_CONFIGURATION,
+          enableExperimentalFeatures: ['postpone_start_recording'],
+        })
+        expect(startRecordingSpy).not.toHaveBeenCalled()
+      })
+
+      it('does not start recording before the page "load"', () => {
+        const { triggerOnLoad } = mockDocumentReadyState()
+        rumRecorderPublicApi.init(DEFAULT_INIT_CONFIGURATION)
+        expect(startRecordingSpy).not.toHaveBeenCalled()
+        triggerOnLoad()
+        expect(startRecordingSpy).toHaveBeenCalled()
+      })
     })
 
-    it('does not start recording when calling init() with the feature "postpone_start_recording"', () => {
-      rumRecorderPublicApi.init({
+    describe('when tracking views manually', () => {
+      const MANUAL_VIEWS_CONFIGURATION = {
         ...DEFAULT_INIT_CONFIGURATION,
-        enableExperimentalFeatures: ['postpone_start_recording'],
+        enableExperimentalFeatures: ['view-renaming'],
+        trackViewsManually: true,
+      }
+      it('starts RUM when initial view is started', () => {
+        expect(startRumSpy).not.toHaveBeenCalled()
+        rumRecorderPublicApi.init(MANUAL_VIEWS_CONFIGURATION)
+        expect(startRumSpy).not.toHaveBeenCalled()
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        ;(rumRecorderPublicApi as any).startView()
```

Comment:
What about tweaking the `rumRecorderPublicApi` variable type instead of casting to any / disabling eslint?
```ts
let rumRecorderPublicApi: RumRecorderPublicApi & { startView(name: string) }
```

Reply:
I have not put too much though about it since we will remove the cast when we will remove the feature flag.
Do you think it still worth to improve it?

Reply:
Ok no problem